synapse/handlers/federation_event.py

```diff
@@ -581,6 +581,13 @@ async def update_state_for_partial_state_event(
                     event.event_id,
                 )
                 return
+
+            # since the state at this event has changed, we should now re-evaluate
+            # whether it should have been rejected. We must already have all of the
+            # auth events (from last time we went round this path), so there is no
+            # need to pass the origin.
+            await self._check_event_auth(None, event, context)
+
```

Comment:
Can we now reject previously unrejected events? If so, we might want to update the `rejections` table or add a todo to do so.

Reply:
yeah; I'm working on that currently.

Reply:
(this is https://github.com/matrix-org/synapse/issues/13006)